actions/bump-version/action.yaml

```diff
@@ -33,12 +33,18 @@ runs:
     - name: Bump version
       id: bump-version
       run: |
-        parameters=(--no-commit --no-tag ${{ inputs.release-type }})
+        parameters=(--no-commit --allow-dirty --no-tag ${{ inputs.release-type }})
         echo "old-version=$(python -c "from configparser import ConfigParser; cfg = ConfigParser(); cfg.read('.bumpversion.cfg'); print(cfg['bumpversion']['current_version'])")" >> "$GITHUB_OUTPUT"
-        if [ -n "${{ inputs.new-version }}" ]; then
```

Comment:
The option to set a new version was nice before. Ideally we can keep it around and it can just be the highest priority case in the if conditions. If new-version is not set as an input, then it makes sense to use our new logic